src/Appwrite/Event/Validator/Event.php

```diff
@@ -40,6 +45,12 @@ public function isValid($value): bool
          * Identify all sections of the pattern.
          */
         $type = $parts[0] ?? false;
+            
+        if ($type == 'functions') {
+            $this->message = 'Triggering a function on a function event is not allowed.';
+            return false;
```

Comment:
This is one approach. 

Alternatively, we can remove the event from:

https://github.com/appwrite/appwrite/blob/0765aac5d27cb7f39e839f85bc6eada87a011cc7/app/config/events.php#L195-L236

Reply:
If we remove the functions events from the events.php file we will also disable it for webhooks and realtime.